spec.html

```diff
@@ -39591,20 +39719,13 @@ <h1>%TypedArray%.prototype.sort ( _comparefn_ )</h1>
           1. Let _len_ be _obj_.[[ArrayLength]].
           1. NOTE: The following closure performs a numeric comparison rather than the string comparison used in <emu-xref href="#sec-array.prototype.sort"></emu-xref>.
           1. Let _SortCompare_ be a new Abstract Closure with parameters (_x_, _y_) that captures _comparefn_ and performs the following steps when called:
-            1. Assert: _x_ is a Number and _y_ is a Number, or _x_ is a BigInt and _y_ is a BigInt.
-            1. If _comparefn_ is not *undefined*, then
-              1. Let _v_ be ? ToNumber(? Call(_comparefn_, *undefined*, « _x_, _y_ »)).
-              1. If _v_ is *NaN*, return *+0*<sub>𝔽</sub>.
-              1. Return _v_.
-            1. If _x_ and _y_ are both *NaN*, return *+0*<sub>𝔽</sub>.
-            1. If _x_ is *NaN*, return *1*<sub>𝔽</sub>.
-            1. If _y_ is *NaN*, return *-1*<sub>𝔽</sub>.
-            1. If _x_ &lt; _y_, return *-1*<sub>𝔽</sub>.
-            1. If _x_ > _y_, return *1*<sub>𝔽</sub>.
-            1. If _x_ is *-0*<sub>𝔽</sub> and _y_ is *+0*<sub>𝔽</sub>, return *-1*<sub>𝔽</sub>.
-            1. If _x_ is *+0*<sub>𝔽</sub> and _y_ is *-0*<sub>𝔽</sub>, return *1*<sub>𝔽</sub>.
-            1. Return *+0*<sub>𝔽</sub>.
-          1. Return ? SortIndexedProperties(_obj_, _len_, _SortCompare_).
+            1. Return ? CompareTypedArrayElements(_x_, _y_, _comparefn_).
+          1. Let _sortedList_ be ? SortIndexedProperties(_obj_, _len_, _SortCompare_, ~read-through-holes~).
+          1. Let _j_ be 0.
+          1. Repeat, while _j_ &lt; _len_,
+            1. Perform ! Set(_obj_, ! ToString(𝔽(_j_)), _sortedList_[_j_], *true*).
+            1. Set _j_ to _j_ + 1.
+          1. Return _obj_.
         </emu-alg>
         <emu-note>
           <p>Because *NaN* always compares greater than any other value, *NaN* property values always sort to the end of the result when _comparefn_ is not provided.</p>
```

Comment:
Similarly, this PR moves the steps that make `*NaN*` always compare greater than any other value. So maybe:
```suggestion
          <p>Because *NaN* always compares greater than any other value (see CompareTypedArrayElements), *NaN* property values always sort to the end of the result when _comparefn_ is not provided.</p>
```